Add documentation for Notion connector (#2151)
elastic/connectors: docs/reference/notion.md

# Notion Connector

The [Elastic Notion connector](../connectors/sources/notion.py) is built with the Elastic connectors Python framework and is available as a self-managed [connector client](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html).

## Availability and prerequisites

This connector is available as a **connector client** from the **Python connectors framework**. To use this connector, satisfy all [connector client requirements](https://www.elastic.co/guide/en/enterprise-search/master/build-connector.html).

This connector is in **beta** and is subject to change. The design and code is less mature than official GA features and is being provided as-is with no warranties. Beta features are not subject to the support SLA of official GA features.

## Usage

> To use this connector as a **connector client**, use the **Notion** tile from the connectors list **Customized connector** workflow.

Reviewer: ```suggestion
To use this connector as a **connector client**, select the **Notion** tile when creating a new connector under **Search -> Connectors**.
```

For additional operations, see [Usage](https://www.elastic.co/guide/en/enterprise-search/master/connectors-usage.html).

## Connecting to Notion

> To connect to Notion, the user needs to [create an Internal Integration](https://www.notion.so/help/create-integrations-with-the-notion-api#create-an-internal-integration) for their Notion workspace that can access resources using the Internal Integration Secret Token. Further, configure the Integration with following settings:

Reviewer: ```suggestion
To connect to Notion, the user needs to [create an internal integration](https://www.notion.so/help/create-integrations-with-the-notion-api#create-an-internal-integration) for their Notion workspace, which can access resources using the Internal Integration Secret Token. Configure the Integration with following settings:
```

> 1. Users need to grant `READ` permission for content, comment and user capabilities for that integration from the Capabilities tab.

Reviewer: ```suggestion
1. Users must grant `READ` permission for content, comment and user capabilities for that integration from the Capabilities tab.
```

> 2. Users have to manually [add the integration as a connection](https://www.notion.so/help/add-and-manage-connections-with-the-api#add-connections-to-pages) in each page and database that needs to be retrieved.

Reviewer: ```suggestion
2. Users must manually [add the integration as a connection](https://www.notion.so/help/add-and-manage-connections-with-the-api#add-connections-to-pages) to each page and database that needs to be retrieved.
```

> After completion, use the following configuration parameters to configure the connector.

Reviewer: ```suggestion
Once completed, use the following parameters to configure the connector.
```

## Configuration

### Configure Notion connector

> The following configuration fields are required:

Reviewer: ```suggestion
Note the following configuration fields:
```

#### `Notion Secret Key`  (required)

> Secret Token assigned to your Integration, for the particular workspace. Example:

Reviewer: ```suggestion
Secret token assigned to your integration, for a particular workspace. Example:
```

- `zyx-123453-12a2-100a-1123-93fd09d67394`

#### `Databases`  (required)

> Comma-separated list of database names that would be fetched by the connector. If the value is `*`, connector will fetch all the databases available in the workspace. Example:

Reviewer: ```suggestion
Comma-separated list of database names to be fetched by the connector. If the value is `*`, connector will fetch all the databases available in the workspace. Example:
```

> - `databse1, database2`

Reviewer: ```suggestion
- `database1, database2`
- `*`
```

#### `Pages`  (required)

> Comma-separated list of page names that would be fetched by the connector. If the value is `*`, connector will fetch all the pages available in the workspace. Example:

Reviewer: ```suggestion
Comma-separated list of page names to be fetched by the connector. If the value is `*`, connector will fetch all the pages available in the workspace. Examples:
```

> - `Page1, Page2`

Reviewer: ```suggestion
- `*`
- `Page1, Page2`
```

This connector is in **beta** and is subject to change. The design and code is less mature than official GA features and is being provided as-is with no warranties. Beta features are not subject to the support SLA of official GA features.

## Usage

To use this connector as a **connector client**, select the **Notion** tile when creating a new connector under **Search -> Connectors**.

For additional operations, see [Usage](https://www.elastic.co/guide/en/enterprise-search/master/connectors-usage.html).

## Connecting to Notion

To connect to Notion, the user needs to [create an internal integration](https://www.notion.so/help/create-integrations-with-the-notion-api#create-an-internal-integration) for their Notion workspace, which can access resources using the Internal Integration Secret Token. Configure the Integration with following settings:

1. Users must grant `READ` permission for content, comment and user capabilities for that integration from the Capabilities tab.

> 2. Users must manually [add the integration as a connection](https://www.notion.so/help/add-and-manage-connections-with-the-api#add-connections-to-pages) to each page and database that needs to be retrieved.

Reviewer 2: This is aligned with the Notion documentation, and seems to be one of their built in limitations (along with no timestamps returned and no DLS support). It would be great if inheriting is an option - @praveen-elastic could you please check whether Sean's suggestion is feasible?

> 2. Users must manually [add the integration as a connection](https://www.notion.so/help/add-and-manage-connections-with-the-api#add-connections-to-pages) to each page and database that needs to be retrieved.

Author: I checked and found that sub-pages are able to inherit the connections of parent page regardless of whether the sub-page was present before adding the connection or the sub-page is created after the connection is added to the parent page.

> 2. Users must manually [add the integration as a connection](https://www.notion.so/help/add-and-manage-connections-with-the-api#add-connections-to-pages) to each page and database that needs to be retrieved.

Author: I've updated the statement specifying that the connection needs to be added only in the top-level pages

- `database1, database2`
- `*`

#### `Pages`  (required)

Comma-separated list of page names to be fetched by the connector. If the value is `*`, connector will fetch all the pages available in the workspace. Examples:

- `*`
- `Page1, Page2`

#### `Index Comments`

Toggle to enable fetching and indexing of comments from the Notion workspace for the configured pages, databases and the corresponding child blocks. Default value is `False`.

### Content Extraction

Refer to [content extraction](https://www.elastic.co/guide/en/enterprise-search/current/connectors-content-extraction.html) in the official docs.

## Documents and syncs

The connector syncs the following objects and entities:
- **Pages**
    - Includes metadata such as page name, id, last updated time, etc.
- **Blocks**
    - Includes metadata such as title, type, id, content (in case of file block), etc.
- **Databases**
    - Includes metadata such as name, id, records, size, etc.
- **Users**
    - Includes metadata such as name, id, email address, etc.
- **Comments**
    - Includes the content and metadata such as id, last updated time, created by, etc.

*NOTE*:

> - Files bigger than 10 MB won't be extracted.

Author: Sure, adding it to the to-do list.

- Permissions are not synced. **All documents** indexed to an Elastic deployment will be visible to **all users with access** to that Elastic Deployment.

## Sync rules

[Basic sync rules](https://www.elastic.co/guide/en/enterprise-search/current/sync-rules.html#sync-rules-basic) are identical for all connectors and are available by default.

## Advanced Sync Rules

> Advanced sync rules are not available for this connector in the present version.

Reviewer 3: Will there be a follow-up PR to update this? I believe the design document I'd seen had planned for Advanced Sync Rules
